ragna/deploy/_api/core.py

```diff
@@ -37,7 +38,11 @@ def get_component(display_name: str) -> Component:
 
         return component
 
-    app = FastAPI(title="ragna", version=ragna.__version__)
+    app = FastAPI(
+        title="ragna",
+        version=ragna.__version__,
+        root_path=os.environ.get("RAGNA_API_ROOT_PATH", ""),
```

Comment:
IMO, this should be a config option. You just need to put

```python
root_path: str = ""
```

into

https://github.com/Quansight/ragna/blob/c08a223e5eb6450681cabd7cfd5843c30dbe8d60/ragna/deploy/_config.py#L55-L62

That way you can set it through the config file but also through the `RAGNA_API_ROOT_PATH` environment variable.